test/perf_memcpy_memset_setTensor.cpp

```diff
@@ -0,0 +1,142 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "miopen/tensor_ops.hpp"
+#include "tensor_holder.hpp"
+#include "gtest/conv_tensor_gen.hpp"
+#include "get_handle.hpp"
+
+#include <chrono>
+
+#define TEST_SAMPLES 10
+
+const std::vector<size_t> GetInput() { return {4, 4, 1115, 1115}; }
+
+// set the gpu_ptr(device)
+template <typename T>
+long long testSetTensor(void* gpu_ptr, const tensor<T>& input)
+{
+    float zero               = 0.f;
+    auto&& handle            = get_handle();
+    long long total_durtaion = 0;
+
+    for(int i = 0; i < TEST_SAMPLES; ++i)
+    {
+        auto start = std::chrono::high_resolution_clock::now();
+        SetTensor(handle, input.desc, gpu_ptr, &zero);
+        // End time
+        auto end = std::chrono::high_resolution_clock::now();
+
+        // Calculating total duration
+        total_durtaion +=
+            std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+    }
+    return total_durtaion / TEST_SAMPLES;
+}
+
+// copy input data (host) to gpu_ptr(device)
+template <typename T>
+long long testHIPMemcpy(void* gpu_ptr, const tensor<T>& input)
+{
+    long long total_durtaion = 0;
+    for(int i = 0; i < TEST_SAMPLES; ++i)
+    {
+        auto start        = std::chrono::high_resolution_clock::now();
+        auto ret_cpy_hTod = hipMemcpy(static_cast<void*>(gpu_ptr),
+                                      input.data.data(),
+                                      sizeof(T) * input.data.size(),
+                                      hipMemcpyHostToDevice);
+        auto end          = std::chrono::high_resolution_clock::now();
+
+        if(ret_cpy_hTod != hipSuccess)
+        {
+            std::cout << "\n\nhipMemcpy from host to device : Error code :" << ret_cpy_hTod
+                      << "\n\n";
+            exit(1);
+        }
+        total_durtaion +=
+            std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+    }
+    return total_durtaion / TEST_SAMPLES;
+}
+
+// set the gpu_ptr(device)
+template <typename T>
+long long testHIPMemset(void* gpu_ptr, const tensor<T>& input)
+{
+    long long total_durtaion = 0;
+    for(int i = 0; i < TEST_SAMPLES; ++i)
+    {
+        auto start = std::chrono::high_resolution_clock::now();
+        auto ret_cpy_hTod =
+            hipMemset(static_cast<void*>(gpu_ptr), 0, sizeof(T) * input.data.size());
```

Comment:
```suggestion
            hipMemset(gpu_ptr, 0, sizeof(T) * input.data.size());
```
`gpu_ptr` is already "void*"